hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -42,173 +38,241 @@ object HealthTransformations {
-        case ("eye_other", categorical) =>
+        else None
+      else None
+    } else None
+    val nonCongenital: Option[HlesHealthCondition] = if (healthCondition.nonCongenital.isDefined) {
+      if (rawRecord.getBoolean(s"hs_${healthCondition.nonCongenital.get._2}_yn"))
+        // non-congenital other
+        if (conditions.apply(healthCondition.condition) == -1L)
           createHealthConditionRow(
             rawRecord,
-            "dx_eye_other",
-            eyeDiseaseCondition,
-            categorical,
-            conditionOtherDescription = rawRecord.getOptional("hs_dx_eye_other_spec")
+            healthCondition.nonCongenital.get._1,
+            conditionTypes.apply(healthCondition.conditionType),
+            conditions.apply(healthCondition.condition),
+            conditionOtherDescription =
+              rawRecord.getOptional(s"hs_${healthCondition.nonCongenital.get._1}_spec")
           )
-        case ("blind", categorical) => {
+        // specific non-congenital blindness case
+        else if (healthCondition.condition == "blind") {
           val isCauseKnown = rawRecord.getBoolean("hs_dx_eye_cause_yn")
           val conditionCause =
             if (isCauseKnown) rawRecord.getOptionalNumber("hs_dx_eye_cause") else None
           createHealthConditionRow(
             rawRecord,
-            "dx_blind",
-            eyeDiseaseCondition,
-            categorical,
+            healthCondition.nonCongenital.get._1,
+            conditionTypes.apply(healthCondition.conditionType),
+            conditions.apply(healthCondition.condition),
             conditionCause = conditionCause,
             conditionCauseOtherDescription =
-              if (isCauseKnown && (conditionCause.getOrElse(None) == 98))
+              if (isCauseKnown && (conditionCause.getOrElse(false) == 98))
                 rawRecord.getOptional("hs_dx_eye_cause_other")
               else None
           )
         }
-        case (disease, categorical) =>
-          createHealthConditionRow(rawRecord, s"dx_${disease}", eyeDiseaseCondition, categorical)
-      }
+        // non congenital general case
+        else if (rawRecord.getBoolean(s"hs_${healthCondition.nonCongenital.get._1}"))
+          createHealthConditionRow(
+            rawRecord,
+            healthCondition.nonCongenital.get._1,
+            conditionTypes.apply(healthCondition.conditionType),
+            conditions.apply(healthCondition.condition)
+          )
+        else None
+      else None
     } else None
+    Iterable.concat(congenital, nonCongenital)
+  }
 
-  ////////////////////////////////
-  //// condition categoricals ////
-  ////////////////////////////////
-  val eyeDiseaseCondition = 1L
-  val infectiousDiseaseCondition = 50L
+  val healthConditions: List[HealthCondition] = List(
+    // eye conditions
+    HealthCondition(
+      "eye",
+      "blind",
+      Some(("cg_eye_blind", "cg_eye_disorders")),
+      Some(("dx_blind", "dx_eye"))
```

Comment:
Some of this paren-ing might look better using the `->` syntax
```suggestion
      Some("dx_blind" -> "dx_eye")
```

Reply:
But no pressure if the way it is now is preferred

Reply:
I'd very much prefer to not go through and change all the commas to arrows 😅